File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //The text
    string text;
    //The sum of the letters
    int letsum;
    //The sum of the words
    int wordsum;
    //The sum of the sentences
    int sensum;
    //Get the text
    text = get_string("Your text, sir? \n");

    letsum = count_letters(text);

    wordsum = count_words(text);

    sensum = count_sentences(text);

    //Avarage number of letter per 100 words
    double l = (float) letsum / (float) wordsum * 100;
    //Avarage number of sentences per 100 words
    double s = (float) sensum / (float) wordsum * 100;
    //calculate the index with the formula and round the number for get the grade
    int index = (int) round(0.0588 * l - 0.296 * s - 15.8);
    //Print the right grade
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}
//count the total letters of the text
int count_letters(string text)
{
    //counter
    int i = 0;
    //Sum of all the letters of the text
    int letsum = 0;
    //Count all the letters of the text
    while (text[i] != '\0')
    {

        if (isalpha(text[i]))
        {
            letsum++;
            i++;
        }
        else
        {
            i++;
        }

    }
    return letsum;
}
//count the total words of the text
int count_words(string text)
{
    //counter
    int i = 0;
    //Sum of all the words of the text
    int wordsum = 1;
    //Count all the words of the text
    while (text[i] != '\0')
    {

        if (isspace(text[i]))
        {
            wordsum++;
            i++;
        }
        else
        {
            i++;
        }

    }
    return wordsum;
}
//count the total sentences of the text
int count_sentences(string text)
{
    //counter
    int i = 0;
    //Sum of all the sentences of the text
    int sensum = 0;
    //Count all the sentences of the text
    while (text[i] != '\0')
    {

        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sensum++;
            i++;
        }
        else
        {
            i++;
        }

    }
    return sensum;
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Your text, sir? ")

    word_sum = count_words(text)
    # Avarage number of letter per 100 words
    l = count_letters(text) / word_sum * 100
    # Avarage number of setences per 100 words
    s = count_setences(text) / word_sum * 100
    # Get the index with the formula and round the number for get the grade
    index = int(round(0.0588 * l - 0.296 * s - 15.8))

    # Print the right Grade
    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


# Functions that count all the Letters, Setences and words in the text

def count_letters(text):
    letter_counter = 0
    for i in text:
        if i.isalpha():
            letter_counter += 1
    return letter_counter


def count_words(text):
    words_counter = 1
    for i in text:
        if i.isspace():
            words_counter += 1
    return words_counter


def count_setences(text):
    setences_counter = 0
    for i in text:
        if i in [".", "?", "!"]:
            setences_counter += 1
    return setences_counter


# Call main
main()
